rmd_files/RNAseq_report.Rmd

```diff
@@ -3558,28 +3566,57 @@ dragen.fusions$geneB_dna_support <- "-"
-  fusions$Dragen <- c(rep("-", nrow(fusions)))
-  fusions$split_reads <- fusions$split_readsA + fusions$split_readsB
-  fusions$soft_clipped_reads <- c(rep("-", nrow(fusions)))
-  fusions$score <- c(rep("-", nrow(fusions)))
-  
-  ##### Re-order columns
-  fusions <- fusions %>% dplyr::relocate(split_reads, .before = split_readsA)
-  fusions <- fusions %>% dplyr::relocate(soft_clipped_reads, .before = confidence)
-  fusions <- fusions %>% dplyr::relocate(score, .before = FGID)
+  #####  Dragen's fusion format version 3.9.3
+  if ( all(c("GeneALocation", "GeneBLocation", "NumSplitReads","NumSoftClippedReads", "Score") %in% colnames(dragen.fusions)) ) {
+
+    ##### Add column with Dragen fusions
+    dragen.fusions$fusion <- paste(dragen.fusions$geneA, dragen.fusions$geneB, sep="__")
+    fusions$Dragen <- c(rep("-", nrow(fusions)))
+    fusions$split_reads <- fusions$split_readsA + fusions$split_readsB
+    fusions$soft_clipped_reads <- c(rep("-", nrow(fusions)))
+    fusions$score <- c(rep("-", nrow(fusions)))
+    
+    ##### Re-order columns
+    fusions <- fusions %>% dplyr::relocate(split_reads, .before = split_readsA)
+    fusions <- fusions %>% dplyr::relocate(soft_clipped_reads, .before = confidence)
+    fusions <- fusions %>% dplyr::relocate(score, .before = FGID)
+    
+    ##### Loop through Dragen results, mark fusions detected by both tools. For those detected only by Dragen adapt results format to Arriba results
+    for ( i in 1:nrow(dragen.fusions) ) {
+      
+      if ( !is.na(match(dragen.fusions$fusion[i], paste(fusions$geneA, fusions$geneB, sep="__"))) ) {
+        fusions$Dragen[ match(dragen.fusions$fusion[i], paste(fusions$geneA, fusions$geneB, sep="__")) ] <- "Yes"
+        dragen.fusions[ i, ] <-  rep("-", ncol(dragen.fusions))
+      } else {
+        fusions <- rbind(fusions, data.frame(geneA=dragen.fusions$geneA[i],geneB=dragen.fusions$geneB[i], breakpointA=dragen.fusions$LeftBreakpoint[i], breakpointB=dragen.fusions$RightBreakpoint[i], siteA=dragen.fusions$GeneALocation[i], siteB=dragen.fusions$GeneBLocation[i], type="-", split_reads=dragen.fusions$NumSplitReads[i], split_readsA="-", split_readsB="-", discordant_mates="-", soft_clipped_reads=dragen.fusions$NumSoftClippedReads[i], confidence="-", score=dragen.fusions$Score[i], FGID=dragen.fusions$FGID[i], reported_fusion=dragen.fusions$reported_fusion[i], reported_fusion_geneA=dragen.fusions$reported_fusion_geneA[i], reported_fusion_geneB=dragen.fusions$reported_fusion_geneB[i], effector_gene=dragen.fusions$effector_gene[i], fusions_cancer=dragen.fusions$fusions_cancer[i], geneA_dna_support="-", geneB_dna_support="-", Arriba="-", Dragen="Yes" ))
+      }
+    }
   
-  ##### Loop through Dragen results, mark fusions detected by both tools. For those detected only by Dragen adapt results format to Arriba results
-  for ( i in 1:nrow(dragen.fusions) ) {
+  #####  Dragen's fusion format version 3.7.5
+  } else if ( "Score" %in% colnames(dragen.fusions) ) {
```

Comment:
As `Score` exists in output of both Dragen versions - should we use `else` instead of `else if` assuming if it's not 3.9.3 then it'll be one of older Dragen versions that only had `FusionGene	Score	LeftBreakpoint	RightBreakpoint	ReadNames`

Reply:
That's correct, we could do that to simplify the code. Good spot